New package: Dedispersion v0.0.1

JuliaRegistrator opened the pull request:
- Registering package: Dedispersion
- Repository: https://github.com/astrogewgaw/Dedispersion.jl
- Created by: @astrogewgaw
- Version: v0.0.1
- Commit: 8aeb094d3dec1172e969545c9be4bdb9e7dcb9e5
- Reviewed by: @astrogewgaw
- Reference: https://github.com/astrogewgaw/Dedispersion.jl/commit/8aeb094d3dec1172e969545c9be4bdb9e7dcb9e5#commitcomment-140190092
- Description: Dedispersion algorithms.
<!-- bf0c69308befbd3ccf2cc956ac8a46712550b79fc9bfb5e4edf8f833f05f4c18b06eddad8845b45beb9f45c2b8020dd6c3e7184cf97bbefe178f40b3e98f1e972f153b75f3e31886e9d791a1ee012e2ae608c1b814eb3c9878083ae32e0bc012e9991dbc162e4a1f2b9c77c04b11e8cd1ac9498604a94c18e2f67167e476eea5e0a2f9ccdad959ebce838313f07162b29345c1d0d2f9646450e7ea645e44e8af849a65bba50087d98143f2e2fb649580e2283482e04f8d6d30c07501aaf81128323e980065abfacb0c2c668de4b70da9efe1b20f1263d4b1553368e056a5b4b9 -->

goerz commented:
Thank you for contributing your package! Please make sure to add a little bit of documentation before registering, though. At a minimum, this would be an explanation in the README what "Dedispersion" is and a usage example.

astrogewgaw commented:
> Thank you for contributing your package! Please make sure to add a little bit of documentation before registering, though. At a minimum, this would be an explanation in the README what "Dedispersion" is and a usage example.

Hi @goerz! Yes, no worries, the documentation is forthcoming.

astrogewgaw commented:
> Thank you for contributing your package! Please make sure to add a little bit of documentation before registering, though. At a minimum, this would be an explanation in the README what "Dedispersion" is and a usage example.

Also, @goerz, I think your comment might be blocking this PR from being merged. Could you add a `[noblock]` to it?

goerz commented:
[noblock] Sorry, I don’t see that you added the missing documentation.

> the documentation is forthcoming

The registration can’t proceed until the package has a reasonable minimum amount of documentation that makes it useable to the general public.

astrogewgaw commented:
[noblock]

> [noblock] Sorry, I don’t see that you added the missing documentation.
> 
> > the documentation is forthcoming
> 
> The registration can’t proceed until the package has a reasonable minimum amount of documentation that makes it useable to the general public.

@goerz First of all, this package is a meta-package. All of its constituent packages have their documentation. Second of all, there is no requirement for a "minimum" amount of documentation in Julia's docs about making and registering new packages. I have already passed all the criteria for auto-merging, and I should be allowed to do so. When I said the documentation is forth-coming, I meant I will take up it up when I have more time. In the meantime, I would like other people to be able to use this package, and so would like it to make it to the General registry without any further ado.

goerz commented:
> Second of all, there is no requirement for a "minimum" amount of documentation in Julia's docs about making and registering new packages.

Sorry, that is not true: https://github.com/JuliaRegistries/General?tab=readme-ov-file#should-i-register-my-package-now

This guideline is not checked *automatically*, because there's some judgement required that a bot can't easily do, but that doesn't mean it's not enforced. 

See also https://discourse.julialang.org/t/when-does-it-make-sense-to-post-a-new-package-to-the-julia-repository/108701/4

Please note that this is a *very* low bar. All that is required is for the README to give some idea about the purpose of the package and for how people might be able to use it. It should not take you more than 15 or 20 minutes to write a README. A "full" documentation (with Documenter or something like that) is absolutely something that can come after registration.

Packages without *any* meaningful documentation are not yet ready for registration and will not be merged.

[noblock]

JuliaRegistrator committed:
New package: Dedispersion v0.0.1

UUID: fc859fcf-df02-4415-b05f-e38a41bf0974
Repo: https://github.com/astrogewgaw/Dedispersion.jl.git
Tree: 46d9e4ad1d550c6b7ac3ed139fb6efc3c45ad8fd

Registrator tree SHA: 17aec322677d9b81cdd6b9b9236b09a3f1374c6a

astrogewgaw commented:
@goerz My apologies for getting back so late. I agree with you that adding a README is indeed a very low bar. At the time, I was terribly busy, and thought it might be a good idea to push for the registration of the package first, and worry about the documentation later. However, in retrospect, I agree that a README should have been added nonetheless. I finally found some time, and added more content to the README as suggested. Since this is a meta-package, there is no usage specified, since for that users can just go to the individual component packages. Let me know if this up to snuff, and if so, do unblock the PR since otherwise it will close in 7 days. Cheers!

goerz commented:
[noblock] Thanks! That looks good! You’ll have to add `[noblock]` to your own comment as well

astrogewgaw commented:
[noblock] Thanks @goerz!
